lib/rules/no-is-mounted.js

```diff
@@ -38,7 +39,10 @@ module.exports = {
         if (callee.type !== 'MemberExpression') {
           return;
         }
-        if (callee.object.type !== 'ThisExpression' || callee.property.name !== 'isMounted') {
+        if (
+          callee.object.type !== 'ThisExpression'
+          && (!('name' in callee.property) || callee.property.name !== 'isMounted')
+        ) {
```

Comment:
This caused a regression, see https://github.com/jsx-eslint/eslint-plugin-react/issues/3819

It should always return when `callee.object.type` is not `'ThisExpression'` – no matter if `'name'` is missing in `callee.property` or not.

The functionality before this PR would be equivalent to this type checker compliant check:

```js
        if (
          callee.object.type !== 'ThisExpression' ||
          !('name' in callee.property) ||
          callee.property.name !== 'isMounted'
        ) {
```

Reply:
PR with said fix: https://github.com/jsx-eslint/eslint-plugin-react/pull/3821